common/src/main/scala/org/apache/comet/CometConf.scala

```diff
@@ -263,6 +263,12 @@ object CometConf extends ShimCometConf {
       .booleanConf
       .createWithDefault(false)
 
+  val COMET_REPLACE_SMJ: ConfigEntry[Boolean] =
+    conf(s"$COMET_EXEC_CONFIG_PREFIX.replaceSortMergeJoin")
+      .doc("Whether to replace SortMergeJoin with ShuffledHashJoin for improved performance.")
+      .booleanConf
+      .createWithDefault(true)
```

Comment:
I think we should have a default value as false for stablility. Spark decides to use SMJ for some reasons including data statistics. If Spark thinks SHJ may not work, I think we better follow it except for explicitly asking by users.

Reply:
The other accelerators (Spark RAPIDS and Gluten) default this to true. Perhaps we should benchmark at large scale factors before and see if we run into any issues?

Reply:
I guess it is okay for the benchmark datasets like TPCDS or TPCH. The cases I worry about is the production ones. But it might be more internal cases.

For OSS, maybe enabling it by default is okay.

Reply:
We should add some more descriptions here to mention the risk.

Reply:
> Perhaps we should benchmark at large scale factors before and see if we run into any issues?

Agreed. (Also, when I wrote SF1 and SF10 I meant 1TB, and 10TB which is really SF 1000 and SF 10000).

Reply:
For this PR, I disabled the feature by default. I created the following PR to enable it by default and update the tests. I will add documentation as part of this PR.

https://github.com/apache/datafusion-comet/pull/1008

Reply:
Here is a new follow on issue for enabling by default:

https://github.com/apache/datafusion-comet/issues/1011